Add pre-commit-hooks.yaml

jugmac00 opened the pull request:
Hi Marius,

this PR adds a config file for pre-commit, so this package can be used with pre-commit.

If you accept the PR, it would be awesome to have a new release soon.

Jürgen

mgedmin commented on .pre-commit-hooks.yaml:
```diff
@@ -0,0 +1,7 @@
+    entry: check-python-versions
+    language: python
+    pass_filenames: false
+    always_run: true
```
git shows a whitespace error (missing newline at EOF)

jugmac00 replied:
This was intentionally left out to match the YAML style in `check-manifest` :-)

mgedmin replied:
Huh?  I don't see any git whitespace errors in https://github.com/mgedmin/check-manifest/pull/102/files

jugmac00 replied:
Hm... GitHub does not show the emtpy line....

https://github.com/mgedmin/check-manifest/blob/master/.pre-commit-hooks.yaml

mgedmin replied:
What empty line?  There's no empty line.

I think we're seeing a mismatch between your text editor of choice and the rest of the world.  On Unix every line, including the last line in a file, is supposed to end in a line terminator (\n).  Some other operating systems (DOS/Windows) use a line _separator_ instead of a line _terminator_, so the last line is not followed by anything.

I think Emacs is one text editor that flaunts Unix conventions and shows a phantom empty line at the bottom of every correctly terminated Unix file.  I know that Vim doesn't (and Vim goes out of its way to add the \n after the last line in a file, unless you explicitly set a bunch of options to prevent it).

git tries to enforce Unix conventions and so git diff shows a warning if you try to add a file without the trailing newline.  GitHub copies this.

jugmac00 commented:
> Would you mind adding a little bit of pre-commit usage instructions to the README? Something like https://github.com/mgedmin/check-manifest/blob/4a23fb56952ec5376007108239e94093a44cd51e/README.rst#version-control-integration

Hm, afaik there is a problem with this approach, similar to the chicken and egg problem.

If I add something like
```
  - repo: https://github.com/jugmac00/check-python-versions
    rev: "0.14.3"
    hooks:
    - id: check-python-versions
```
this will error as version "0.14.3" does not contain a hook file.

I could set version to `0.14.4` or `0.15` - but then you should almost immediately make a new release - otherwise this hook will error again - this time for a "pathspec did not match any files known to git" reason.

This is the README block - either tell me which version number I should set or I create a PR after you did a first release with the pre-commit hook.

```
pre-commit integration
----------------------

With `pre-commit <https://pre-commit.com>`_,
``check-python-versions`` can be part of your git-workflow.
Add the following snippet to your ``.pre-commit-config.yaml``.

.. code-block:: yaml

    repos:
    -   repo: https://github.com/mgedmin/check-python-versions
        rev: "0.14.3"
        hooks:
        -   id: check-python-versions
```

mgedmin commented:
I will immediately make a release after merging this.

Let's make the next version 0.15 -- pre-commit integration feels like a feature to me.

The suggested README snippet looks good to me!

mgedmin commented:
Oh! Oh! Oh! If you'd also add a changelog note to CHANGES.rst, I'd appreciate it.

jugmac00 commented:
@mgedmin I added the requested changes - I am not aware what tooling you use for package creation - please be aware, that I updated the change log entry to `0.15 (unreleased)`.

mgedmin commented:
I use a combination of zest.releaser (prerelease/postrelease) and roll-your-own `make release`, which mostly prints out commands for me to copy and paste into a terminal.

mgedmin commented:
check-python-versions 0.15.0 is released to PyPI.

jugmac00 commented:
I have noticed a glitch - the new version is tagged `0.15.0` and in the `README` I wrote `0.15`. I'll prepare a pull request to fix this. Sorry for the misunderstanding!

jugmac00 commented:
#17

mgedmin commented:
Oops!  My fault entirely, I checked the style of the previous version tags while doing the release and changed it to match, without realizing.
